test cases/rust/12 bindgen/meson.build

```diff
@@ -33,69 +37,76 @@ gen = rust.bindgen(
   include_directories : 'include',
 )
 
-# see: https://github.com/mesonbuild/meson/issues/8160
-f = configure_file(
-  input : 'src/main.rs',
-  output : 'main.rs',
-  copy : true,
-)
-
-rust_bin = executable(
-  'rust_bin',
-  [f, gen],
-  link_with : c_lib,
-)
-
-test('main', rust_bin)
-
-# Test a generated header
-gen_h = custom_target(
-  'gen.h',
-  command : [find_program('src/gen_header.py'), '@INPUT@', '@OUTPUT@'],
-  output : 'gen.h',
-  input : 'src/header.h'
-)
-
-gen2_h = custom_target(
-  'other.h',
-  command : [find_program('src/gen_header.py'), '@INPUT@', '@OUTPUT@'],
-  output : 'other.h',
-  input : 'include/other.h'
-)
-
-gen_rs = rust.bindgen(
-  input : [gen_h, gen2_h],
-  output : 'gen.rs',
-)
-
-f = configure_file(
-  input : 'src/main2.rs',
-  output : 'main2.rs',
-  copy : true,
-)
-
-rust_bin2 = executable(
-  'rust_bin2',
-  [f, gen_rs],
-  link_with : c_lib,
-)
-
-test('generated header', rust_bin2)
-
-subdir('sub')
-subdir('dependencies')
-
-gp = rust.bindgen(
-  input : 'src/global-project.h',
-  output : 'global-project.rs',
-)
-
-gp_lib = static_library('gp_lib', 'src/global.c')
-
-gp_exe = executable(
-  'gp_exe',
-  structured_sources(['src/global.rs', gp]),
-  link_with : gp_lib,
-)
-
-test('global and project arguments', gp_exe)
+if not get_option('bad-values')
```

Comment:
Why is this conditional needed? I don't understand it. Shouldn't all the code be resistant to CFLAGS that it doesn't understand as long as those are valid CFLAGS?

Reply:
The problem is that it tries to initiate the c compiler with the bad flag, and then fails. Maybe this level of complexity isn’t worth it compared to just writing a plain unit test

Reply:
The problem seems to be compiler flags that aren't understood by the Clang compiler, fwict. So it should be possible to test it like so:

```
cc_id = cc.get_id()
if cc_id == 'gcc'
  bad_values = ['-mtls-dialect=gnu2']
elif cc_id == 'msvc'
  bad_values = ['/fp:fast']
endif
```

Flags like `-fmath-errno` that are understood by both GCC and Clang do not cause any issues.

Reply:
I pushed a fixup commit that implements this. I tested that the tests fail if I revert the fixes.